src/components/MyAccount/RequestsTab/FreezeButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react"
+import type { Hold, Patron } from "../../../types/accountTypes"
+import { Button } from "@nypl/design-system-react-components"
+
+const FreezeButton = ({ hold, patron }: { hold: Hold; patron: Patron }) => {
+  const [frozen, setFrozen] = useState(hold.frozen)
+  const [isDisabled, setIsDisabled] = useState(false)
+
+  const handleFreezeClick = async () => {
+    // Disabling button while request happens.
+    setIsDisabled(true)
+    if (frozen) {
+      await fetch(
+        `/research/research-catalog/api/account/holds/update/${hold.id}`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            patronId: patron.id,
+            freeze: false,
+            pickupLocation: hold.pickupLocation.code,
+          }),
+        }
+      )
+      setFrozen(false)
+    } else {
+      await fetch(
```

Comment:
I keep going back and forth about this... I'm wondering if it would be worth it to make a generic fetch call that would take a path, method, and body. or maybe the answer is to take those test methods from the [[index]] page and keep those all in a file somewhere. these options objects take up a lot of visual space in the code and don't have anything to do with what's rendered. @EdwinGuzman do you have thoughts on this too?

Reply:
Yes, I agree there should be a function that wraps the fetch call. If it's created for this PR, great, but it can also wait for another PR.

Emma, this is similar to the `apiCall` function in digital-collections. The function for this app may need more arguments though.